src/common/getters.ts

```diff
@@ -212,6 +212,235 @@ export function getPurposes(vc: DatasetWithId): string[] {
+ * ```
+ *
+ * @param accessCredential The Access Credential (Access Grant or Access request)
+ * @returns the value of the custom field with the provided name if it is a boolean, undefined otherwise.
+ * @since unreleased
+ */
+export function getCustomDouble(
+  vc: DatasetWithId,
+  field: URL,
+): number | undefined {
+  const val = Array.from(
+    vc.match(getConsent(vc), namedNode(field.href), null, defaultGraph()),
+  )[0]?.object.value;
+  return deserizalizeDouble(val);
+}
+
+const xmlSchemaTypes = {
+  boolean: namedNode("http://www.w3.org/2001/XMLSchema#boolean"),
+  double: namedNode("http://www.w3.org/2001/XMLSchema#double"),
+  integer: namedNode("http://www.w3.org/2001/XMLSchema#integer"),
+  string: namedNode("http://www.w3.org/2001/XMLSchema#string"),
+} as const;
+
+function castLiteral(lit: Literal): unknown {
+  if (lit.datatype.equals(xmlSchemaTypes.boolean)) {
+    return deserializeBoolean(lit.value);
+  }
+  if (lit.datatype.equals(xmlSchemaTypes.double)) {
+    return deserizalizeDouble(lit.value);
+  }
+  if (lit.datatype.equals(xmlSchemaTypes.integer)) {
+    return deserizalizeInteger(lit.value);
+  }
+  if (lit.datatype.equals(xmlSchemaTypes.string)) {
+    return lit.value;
+  }
+  throw new Error(`Unsupported literal type ${lit.datatype.value}`);
+}
+
+const WELL_KNOWN_FIELDS = [
+  gc.forPersonalData,
+  gc.forPurpose,
+  gc.isProvidedTo,
+  gc.isProvidedToController,
+  gc.isProvidedToPerson,
+  acl.mode,
+  INHERIT,
+];
+
+/**
+ * Reads all the custom fields in the consent section of the provided Access Credential.
+ *
+ * @example
+ * ```
+ * const accessRequest = await issueAccessRequest({...}, {
+ *  ...,
+ *  customFields: new Set([
+ *     {
+ *       key: new URL("https://example.org/ns/customString"),
+ *       value: "custom value",
+ *     },
+ *     {
+ *       key: new URL("https://example.org/ns/customInteger"),
+ *       value: 1,
+ *     },
+ *   ]),
+ * });
+ * const customFields = getCustomFields(accessRequest);
+ * // s is "custom value"
+ * const s = customFields["https://example.org/ns/customString"];
+ * // i is 1
+ * const i = customFields["https://example.org/ns/custominteger"];
+ * ```
+ *
+ * @param accessCredential The Access Credential (Access Grant or Access request)
+ * @returns an object keyed by the custom fields names, associated to their values.
+ * @since unreleased
+ */
+export function getCustomFields(
+  accessCredential: DatasetWithId,
+): Record<string, unknown> {
+  const consent = getConsent(accessCredential);
+  return Array.from(accessCredential.match(consent, null, null, defaultGraph()))
+    .filter(
+      ({ predicate, object }) =>
+        // The Access Grant data model is known, so by elimination any unknown
+        // field is a custom one.
+        !WELL_KNOWN_FIELDS.some((wellKnown) => wellKnown.equals(predicate)) &&
+        // Nested objects are not supported.
+        object.termType === "Literal",
```

Comment:
Needs indentation as a continuation of previous line

Reply:
This was removed in https://github.com/inrupt/solid-client-access-grants-js/pull/1178